File: pocs/temporal-ai-poc/src/worker.ts
import { NativeConnection, Worker } from '@temporalio/worker';
import * as activities from './activities.ts';
import { TASK_QUEUE } from './shared.ts';

const address = process.env.TEMPORAL_ADDRESS ?? 'localhost:7233';

async function run() {
  const connection = await NativeConnection.connect({ address });
  const worker = await Worker.create({
    connection,
    namespace: 'default',
    taskQueue: TASK_QUEUE,
    workflowsPath: new URL('./workflows.ts', import.meta.url).pathname,
    activities,
  });

  console.log(`Worker connected to ${address}, polling task queue "${TASK_QUEUE}"`);
  await worker.run();
}

run().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: pocs/gpt-5.6-sol-model-pipeline/model-pipeline/worker.py
import asyncio
import os

from temporalio.client import Client
from temporalio.worker import Worker

from model_pipeline.training import train_model_batch
from model_pipeline.workflow import TrainModelWorkflow


async def main() -> None:
    client = await Client.connect(os.getenv("TEMPORAL_ADDRESS", "localhost:7233"))
    worker = Worker(client, task_queue="model-training", workflows=[TrainModelWorkflow], activities=[train_model_batch])
    print("Training worker is ready")
    await worker.run()


if __name__ == "__main__":
    asyncio.run(main())
